keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/AdminService.kt

```diff
@@ -25,4 +27,9 @@ class AdminService(
   fun getManagedApplications(): Collection<ApplicationSummary> {
     return repository.getApplicationSummaries()
   }
+
+  fun triggerRecheck(resourceId: String) {
+    log.info("Triggering a recheck for $resourceId by clearing our record of the diff")
+    diffFingerprintRepository.clear(resourceId)
```

Comment:
Just so I understand, the reason we do this by clearing the diff fingerprint is that `UnhappyVeto` uses the diff count to determine whether to veto or allow a resource to be checked?

Reply:
Yep! https://github.com/spinnaker/keel/blob/master/keel-core/src/main/kotlin/com/netflix/spinnaker/keel/veto/unhappy/UnhappyVeto.kt#L56